File: task-management-frontend/src/services/TaskService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8080/api/tasks';

const getAllTasks = () => axios.get(BASE_URL);
const createTask = (task) => axios.post(BASE_URL, task);
const updateTask = (id, task) => axios.put(`${BASE_URL}/${id}`, task);
const updateTaskStatus = (id, status) => axios.put(`${BASE_URL}/update-status/${id}`, { status });
const deleteTask = (id) => axios.delete(`${BASE_URL}/${id}`);

export default {
  getAllTasks,
  createTask,
  updateTask,
  deleteTask,
  updateTaskStatus,  

};

File: task-management-backend/src/main/java/com/task/management/services/TaskService.java
package com.task.management.services;

 
import java.util.List;
import java.util.Optional;

import com.task.management.entities.Task;
 
public interface TaskService {
    List<Task> getAllTasks();
    Optional<Task> getTaskById(Long id);
    Task createTask(Task task);
    Task updateTask(Long id, Task taskDetails);
    void deleteTask(Long id);
    void updateTaskStatus(Long Id, String status);
    List<Task> getAllTasks(String status, String teamMemberName);

}
